vrp-pragmatic/src/format/solution/writer.rs

```diff
@@ -79,6 +90,22 @@ fn write_geo_json<W: Write>(problem: &Problem, solution: &Solution, writer: &mut
     Ok(())
 }
 
+fn write_geo_json_and_pragmatic_json<W: Write>(
+    problem: &Problem,
+    solution: &Solution,
+    metrics: Option<&TelemetryMetrics>,
+    writer: &mut BufWriter<W>,
+) -> Result<(), String> {
+    let pragmatic_solution = create_solution(problem, solution, metrics);
+    let geo_json_solution = create_solution(problem, solution, None);
+    writer.write(b"{\"pragmatic\":").map_err(|err| err.to_string())?;
+    serialize_solution(&pragmatic_solution, writer).map_err(|err| err.to_string())?;
+    writer.write(b", \"geojson\":").map_err(|err| err.to_string())?;
+    serialize_solution_as_geojson(writer, problem, &geo_json_solution).map_err(|err| err.to_string())?;
+    writer.write(b"}").map_err(|err| err.to_string())?;
+    Ok(())
```

Comment:
It would be nice to avoid somehow constructing json manually: it is a bit error prone. I think one way would be to add dedicated method inside `geo_serializer.rs` with the same signature as this one. Internally, this method will use `create_geojson_solution` [1] which returns `FeatureCollection` object. So, now you can have some private helper struct which will just hold both objects and marked `Serializable`, so you can call serde_json on it. Something like that:

```rust
#[derive(Clone, Deserialize, Serialize, PartialEq, Debug)]
struct SolutionAndFeatureCollection {
   pragmatic: Solution,
   geo: FeatureCollection
}
```
(maybe there is a better name for the struct)

Please don't forget to add some unit tests for it.

[1] https://github.com/reinterpretcat/vrp/blob/master/vrp-pragmatic/src/format/solution/geo_serializer.rs#L272